Limit interface scale

AlbyIanna opened the pull request:
### Motivation
The interface scaling of the IDE doesn't handle scaling properly:
- as stated in #1384, it's possible to find yourself in unpredictable situations we're only limiting the scaling when using the stepper input
- same thing happen with the editor font size
- when changing the scaling really fast (e.g.: holding down `CTRL/CMD + =` for a while), the interface seems to 'bounce' from bigger to smaller and viceversa until it's stable

### Change description
- move the interface scaling logic to a separate contribution
- the stepper in the settings dialog and the related commands now uses common values for min/max scale and font size
- when the `Automatic` flag in the settings dialog is **checked**, enable/disable the **'Increase/Decrease Font Size'** menu item and shortcuts when reaching `-60%` (min) and `280%` (max) 
- when the `Automatic` flag in the settings dialog is **unchecked**, enable/disable the **'Increase/Decrease Font Size'** menu item and shortcuts when reaching `8` (min) and `72` (max)

### Other information
Since [Theia doesn't support dynamic menu items](https://github.com/eclipse-theia/theia/issues/446), to achieve this I needed to dispose the menu items and re-register them, and then [update the `mainMenuManager`](https://github.com/arduino/arduino-ide/compare/limit-interface-scale?expand=1#diff-e85455776c63a56847b610b7e290b72e1ffb4bcfb0188d3351d9fe17866c33a2R122) manually, otherwise they wouldn't get updated as expected.

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

kittaakos commented on arduino-ide-extension/src/browser/contributions/interface-scale.ts:
```diff
@@ -0,0 +1,233 @@
+  };
+
+  private currentScale: InterfaceScale.ScaleSettings;
+  private currentSettings: Settings;
```
Can you explain why IDE2 needs to duplicate the `Settings` state twice?

AlbyIanna replied:
One duplicate is needed for debouncing purposes, but you're right, I can achieve the same with only one variable. Changing it.

kittaakos replied:
Thank you!

Please note `isEnabled: () => this.fontScalingEnabled.increase,` and `isEnabled: () => this.fontScalingEnabled.decrease,` have absolutely no effect.

Exactly that's why you're using the `PlaceholderMenuNode` menu item to have it there in a disabled state.

AlbyIanna replied:
I think you're wrong here. Those lines are not meant to disable/enable the menu item, but the related commands, so that it prevents zooming in/out with shortcuts when exceeding limits.

kittaakos replied:
- The increase/decrease font size will trigger `this.updateFontSize('decrease'|'increase')`;
 - The `updateFontSize` method min/max the font size. See [here](https://github.com/arduino/arduino-ide/pull/1502/files#diff-e85455776c63a56847b610b7e290b72e1ffb4bcfb0188d3351d9fe17866c33a2R138-R154).

Where did I overlook your code? Thanks!

kittaakos commented:
Do you plan to get rid of the local `settings` field? Is this PR still in progress? See here: https://github.com/arduino/arduino-ide/pull/1502#discussion_r980907171.

AlbyIanna commented:
> Do you plan to get rid of the local settings field? Is this PR still in progress? See here: https://github.com/arduino/arduino-ide/pull/1502#discussion_r980907171.


@kittaakos  I've already got rid of `currentScale` because it wasn't necessary (see [last commit](https://github.com/arduino/arduino-ide/pull/1502/commits/0048b29ff20912362cf0acf3bac17d079e83cc4a)). As state in this comment &rarr; https://github.com/arduino/arduino-ide/pull/1502#discussion_r980914215, but I can't remove `currentSettings` because it's not only a duplicate, it serves as temporary value in order to make the debouncing work correctly (see how I update the settings [here](https://github.com/arduino/arduino-ide/pull/1502/files#diff-e85455776c63a56847b610b7e290b72e1ffb4bcfb0188d3351d9fe17866c33a2R37))

kittaakos reviewed (commented):
Thank you! I checked the code and left a few remarks.

AlbyIanna commented on arduino-ide-extension/src/browser/contributions/interface-scale.ts:
```diff
@@ -0,0 +1,221 @@
+      this.fontScalingEnabled = newFontScalingEnabled;
+    }
+    this.fontScalingEnabled = newFontScalingEnabled;
+    this.updateSettingsDebounced();
```
It's useless to return a promise here, changing it.

kittaakos reviewed (changes requested):
I have noticed some inconsistency between the disabled/enabled state of the menu item and the current `editor.fontSize` value.

Decrease expected to be disabled here:

![Screen Shot 2022-10-04 at 09 07 56](https://user-images.githubusercontent.com/1405703/193757444-7af2e049-1a9a-4c3d-b638-8f6ce46e9819.png)


Increase expected to be disabled here:

![Screen Shot 2022-10-04 at 09 06 00](https://user-images.githubusercontent.com/1405703/193757467-fb312f7e-809e-4a61-914d-0649ff8d13d1.png)


Decrease expected to be enabled here:

![Screen Shot 2022-10-04 at 09 11 27](https://user-images.githubusercontent.com/1405703/193757493-1f0d58d8-edd7-4ecc-9113-2416cbec2ecc.png)

AlbyIanna commented:
> I have noticed some inconsistency between the disabled/enabled state of the menu item and the current editor.fontSize value.

Thank you @kittaakos, I wasn't registering the menus properly after changing `fontScalingEnabled`. Fixed in [this commit](https://github.com/arduino/arduino-ide/pull/1502/commits/ca87632ad0923ae743cd3147deb1543b12002859).

I also the menu items weren't updating when changing the `autoScaleInterface` setting, so I also fixed that in the [last commit](https://github.com/arduino/arduino-ide/pull/1502/commits/2ec546f8be2bc7b2e6c328cdd4ba4d4876df61ee)

per1234 reviewed (approved):
I verified it fixes https://github.com/arduino/arduino-ide/issues/1384

Thanks Alberto!

kittaakos reviewed (approved):
It's working great. Thank you!
